server/controllers/oauth.py

```diff
@@ -91,6 +112,13 @@ def authorize(*args, **kwargs):
     confirm = request.form.get('confirm', 'no')
     return confirm == 'yes'
 
+@oauth.route('/oauth/code')
+def oauth_code():
```

Comment:
I don't really see the benefit of that - the user should go through the entire flow again to get a new code